File: LoginDagger/app/src/main/java/com/charlye934/loginddagger/http/TwitchModule.java
package com.charlye934.loginddagger.http;

import dagger.Module;
import dagger.Provides;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

@Module
public class TwitchModule {
    private String BASE_URL = "https://api.twitch.tv/kraken/";

    @Provides
    OkHttpClient provideHttpClient(){
        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        return new OkHttpClient.Builder().addInterceptor(interceptor).build();
    }

    @Provides
     Retrofit provideRetrofit(String baseUrl, OkHttpClient client) {
        return new Retrofit.Builder()
                .baseUrl(baseUrl)
                .client(client)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
    }

    @Provides
    TwitchAPI provideTwitchService(){
        return provideRetrofit(BASE_URL, provideHttpClient()).create(TwitchAPI.class);
    }
}

File: LoginDagger/app/src/main/java/com/charlye934/loginddagger/http/TwitchModule.kt
package com.charlye934.loginddagger.http

import dagger.Module
import dagger.Provides
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

@Module
class TwitchModule {
    private val BASE_URL = "https://api.twitch.tv/kraken/"

    @Provides
    fun provideHttpClient(): OkHttpClient{
        val interceptor = HttpLoggingInterceptor()
        interceptor.level = HttpLoggingInterceptor.Level.BODY
        return OkHttpClient.Builder().addInterceptor(interceptor).build()
    }

    @Provides
    fun provideRetrofit(baserUrl:String, client: OkHttpClient): Retrofit {
        return Retrofit.Builder()
                .baseUrl(baserUrl)
                .client(client)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
    }

    @Provides
    fun provideTwitchService(): TwitchAPI {
        return provideRetrofit(BASE_URL, provideHttpClient()).create(TwitchAPI::class.java)
    }
}